fineract-investor/src/main/java/org/apache/fineract/investor/domain/ExternalAssetOwnerRepository.java

```diff
@@ -18,10 +18,17 @@
  */
 package org.apache.fineract.investor.domain;
 
+import java.util.Optional;
+import org.apache.fineract.infrastructure.core.domain.ExternalId;
 import org.springframework.data.jpa.repository.JpaRepository;
 import org.springframework.data.jpa.repository.JpaSpecificationExecutor;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
 
 public interface ExternalAssetOwnerRepository
         extends JpaRepository<ExternalAssetOwner, Long>, JpaSpecificationExecutor<ExternalAssetOwner> {
 
+    @Query("select e from ExternalAssetOwner e where e.externalId = :externalId")
```

Comment:
Note for the future, this could've been done as a Spring Data generated query.

Reply:
fixed